### PR TITLE
improve: update to lantern 0.11

### PR DESCRIPTION
**Pull request by samedii**

### CODE REVIEW

File: {{cookiecutter.repository_name}}/debug/evaluate.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from guild.commands.run import run
+
+
+if __name__ == "__main__":
+    run(["evaluate", "-y", "n_workers=0", "--debug-sourcecode=."])
```
Comment (FelixAbrahamsson) [inline {{cookiecutter.repository_name}}/debug/evaluate.py:5]: Imo we should labels these by default :) E.g. `debug-evaluate`/`debug-train`

File: {{cookiecutter.repository_name}}/debug/train.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from guild.commands.run import run
+
+
+if __name__ == "__main__":
+    run(["train", "-y", "n_workers=0", "n_batches_per_epoch=2", "--debug-sourcecode=."])
```
Comment (FelixAbrahamsson) [inline {{cookiecutter.repository_name}}/debug/train.py:5]: Small `max_epochs` and `batch_size` as well? I normally can't fit full batch on my laptop when I'm prototyping + it runs faster

File: {{cookiecutter.repository_name}}/operations/evaluate.py [hunk 3]
```diff
@@ -57,13 +57,13 @@ def evaluate(config):
...
-    parser.add_argument("--n_workers", default=2, type=int)
+    parser.add_argument("--n_workers", type=int, default=2)
+    parser.add_argument("--cuda", type=bool, choices=[True, False], default=True)
     args = parser.parse_args()
 
     config = vars(args)
     config.update(
-        seed=1,
-        use_cuda=torch.cuda.is_available(),
+        seed=2464,
```
Comment (FelixAbrahamsson) [inline {{cookiecutter.repository_name}}/operations/evaluate.py:66]: Does it matter that the seed is a small (low bit entropy i guess?) number? I like that it's easy to remember if you want to set the seed somewhere else

File: {{cookiecutter.repository_name}}/operations/evaluate.py [hunk 4]
```diff
@@ -38,16 +40,14 @@ def evaluate(config):
...
                 predictions = model.predictions(standardized_images)
-                loss = predictions.loss(examples)
 
-            evaluate_metrics[name]["loss"].update_(loss)
-            evaluate_metrics[name]["accuracy"].update_(examples, predictions)
+            evaluate_metrics[name]["standard"].update_(predictions, examples)
 
         for metric_name, metric in evaluate_metrics[name].items():
-            metric.log(tensorboard_logger, name, metric_name)
+            metric.log_dict(tensorboard_logger, name)
```
Comment (JoarGruneau) [inline {{cookiecutter.repository_name}}/operations/evaluate.py:50]: This looks wrong, you are writing the name to metrics?
If it is correct, maybe better naming would be good so it's easy to read

File: {{cookiecutter.repository_name}}/operations/train.py [hunk 5]
```diff
@@ -73,35 +75,35 @@ def train(config):
             optimizer.zero_grad()
 
             train_metrics["loss"].update_(loss)
-            train_metrics["accuracy"].update_(examples, predictions)
+            train_metrics["standard"].update_(predictions, examples)
```
Comment (FelixAbrahamsson) [inline {{cookiecutter.repository_name}}/operations/train.py:78]: What do we think about the name `standard`? The downside is it's not very descriptive, but I struggle to come up with something that is. `from_predictions_and_examples` is a suggestion but it's a little long

File: {{cookiecutter.repository_name}}/debug/evaluate.py (hunk 1 above)
Comment (samedii) [reply {{cookiecutter.repository_name}}/debug/evaluate.py:5]: What do you mean? Do you want to rename the file?

Comment (samedii) [reply {{cookiecutter.repository_name}}/debug/evaluate.py:5]: Nevermind you mean add labels. I tried it but it didn't work for me. Do you know how to?

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/metrics.py [hunk 6]
```diff
@@ -1,34 +1,34 @@
...
 def evaluate_metrics():
     return dict(
-        loss=lantern.MapMetric(lambda loss: loss.item()),
-        accuracy=lantern.MapMetric(
-            map_fn=lambda examples, predictions: [
-                example.class_name == prediction.class_name
-                for example, prediction in zip(examples, predictions)
-            ],
-            compute_fn=lambda results: np.mean(np.concatenate(results)),
+        standard=(
```
Comment (FelixAbrahamsson) [inline {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/metrics.py:18]: What do we think about the name standard? The downside is it's not very descriptive, but I struggle to come up with something that is. from_predictions_and_examples is a suggestion but it's a little long

File: {{cookiecutter.repository_name}}/operations/train.py [hunk 7]
```diff
@@ -73,35 +75,35 @@ def train(config):
...
 
             for name, metric in train_metrics.items():
-                metric.log(tensorboard_logger, "train", name, epoch)
+                metric.log_dict(tensorboard_logger, "train", epoch)
```
Comment (JoarGruneau) [inline {{cookiecutter.repository_name}}/operations/train.py:81]: same here

File: {{cookiecutter.repository_name}}/operations/train.py [hunk 8]
```diff
@@ -73,35 +75,35 @@ def train(config):
...
                     predictions = model.predictions(standardized_images)
-                    loss = predictions.loss(examples)
 
-                evaluate_metrics[name]["loss"].update_(loss)
-                evaluate_metrics[name]["accuracy"].update_(examples, predictions)
+                evaluate_metrics[name]["standard"].update_(predictions, examples)
 
             for metric_name, metric in evaluate_metrics[name].items():
-                metric.log(tensorboard_logger, name, metric_name, epoch)
+                metric.log_dict(tensorboard_logger, name, epoch)
```
Comment (JoarGruneau) [inline {{cookiecutter.repository_name}}/operations/train.py:98]: and here

File: {{cookiecutter.repository_name}}/operations/evaluate.py (hunk 3 above)
Comment (FelixAbrahamsson) [inline {{cookiecutter.repository_name}}/operations/evaluate.py:66]: Also, shouldn't train and evaluate set the same seeds? Imo evaluate can load the seed from the train config?

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/model/model.py [hunk 9]
```diff
@@ -27,9 +27,13 @@ def __init__(self):
         )
 
     def forward(self, prepared):
-        return architecture.PredictionBatch(logits=self.logits(prepared))
+        return model.PredictionBatch(logits=self.logits(prepared))
```
Comment (FelixAbrahamsson) [inline {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/model/model.py:30]: `self.PredictionBatch`? Or is that bad/confusing?

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/model/prediction.py [hunk 10]
```diff
@@ -83,30 +83,5 @@ def cross_entropy(self, examples):
             self.stack_class_indices(examples),
         )
 
-    def cpu(self):
```
Comment (FelixAbrahamsson) [inline {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/model/prediction.py:86]: Why do we remove this and `detach`?

File: {{cookiecutter.repository_name}}/debug/evaluate.py (hunk 1 above)
Comment (FelixAbrahamsson) [reply {{cookiecutter.repository_name}}/debug/evaluate.py:5]: `--label debug-train` works for me

Comment (FelixAbrahamsson) [reply {{cookiecutter.repository_name}}/debug/evaluate.py:5]: `-l` should work too but didn't seem to work with some version of guild for some reason

File: {{cookiecutter.repository_name}}/operations/evaluate.py (hunk 3 above)
Comment (samedii) [reply {{cookiecutter.repository_name}}/operations/evaluate.py:66]: Why? Seeds don't work that way

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/metrics.py (hunk 6 above)
Comment (samedii) [reply {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/metrics.py:18]: `pairs`?

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/model/model.py (hunk 9 above)
Comment (samedii) [reply {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/model/model.py:30]: It's maybe a little more confusing

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/model/prediction.py (hunk 10 above)
Comment (samedii) [reply {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/model/prediction.py:86]: They are very ugly and they are not used anymore

File: {{cookiecutter.repository_name}}/operations/evaluate.py (hunk 3 above)
Comment (samedii) [reply {{cookiecutter.repository_name}}/operations/evaluate.py:66]: Maybe not. I don't feel comfortable promoting the selection seed=1,2,3,... for different splits etc

File: {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/model/model.py (hunk 9 above)
Comment (FelixAbrahamsson) [reply {{cookiecutter.repository_name}}/{{cookiecutter.package_name}}/model/model.py:30]: I thought it could be less confusing since it's used under the `Model` namespace everywhere else, except where it's defined. But it doesn't really matter much.

**Review by FelixAbrahamsson — APPROVED** [review]
